packages/template-retail-react-app/app/pages/checkout/partials/shipping-options.jsx

```diff
@@ -11,111 +11,237 @@ import {
+                <Text fontWeight="bold" fontSize="md" mb={1}>
+                    {formatMessage(
+                        {
+                            defaultMessage: 'Delivering to {name}',
+                            id: 'shipping_options.label.delivering_to'
+                        },
+                        {
+                            name: `${shipment.shippingAddress.firstName} ${shipment.shippingAddress.lastName}`
+                        }
+                    )}
+                </Text>
+                <Text fontSize="sm" color="gray.600">
+                    {shipment.shippingAddress.address1}, {shipment.shippingAddress.city},{' '}
+                    {shipment.shippingAddress.stateCode} {shipment.shippingAddress.postalCode}
+                </Text>
+            </Box>
+
+            {/* Combined Product Cards and Shipping Options */}
+            <Box
+                border="1px solid"
+                borderColor="gray.200"
+                borderRadius="lg"
+                pt={2}
+                pb={4}
+                px={2}
+                bg="white"
+            >
+                <VStack spacing={4} align="stretch">
+                    {/* Product Cards */}
+                    <ShippingProductCards shipment={shipment} basket={basket} />
+
+                    {/* Shipping Options */}
+                    <ShippingOptionsList
+                        shipment={shipment}
+                        basketId={basketId}
+                        currency={currency}
+                        control={control}
+                    />
+                </VStack>
+            </Box>
+        </VStack>
+    )
+}
+
+ShipmentOptionsWithProducts.propTypes = {
+    shipment: PropTypes.shape({
+        shipmentId: PropTypes.string.isRequired,
+        shippingAddress: PropTypes.shape({
+            firstName: PropTypes.string,
+            lastName: PropTypes.string,
+            address1: PropTypes.string,
+            city: PropTypes.string,
+            stateCode: PropTypes.string,
+            postalCode: PropTypes.string
+        }),
+        shippingMethod: PropTypes.shape({
+            id: PropTypes.string
+        })
+    }).isRequired,
+    basketId: PropTypes.string.isRequired,
+    currency: PropTypes.string.isRequired,
+    control: PropTypes.object.isRequired,
+    basket: PropTypes.shape({
+        productItems: PropTypes.arrayOf(
+            PropTypes.shape({
+                itemId: PropTypes.string.isRequired,
+                shipmentId: PropTypes.string,
+                productName: PropTypes.string,
+                image: PropTypes.string,
+                imageUrl: PropTypes.string,
+                primaryImage: PropTypes.string,
+                images: PropTypes.array,
+                quantity: PropTypes.number,
+                variationValues: PropTypes.object,
+                variations: PropTypes.object
+            })
+        )
+    }).isRequired
+}
 
 export default function ShippingOptions() {
```

Comment:
Maybe we should rename this component to make it clearer
```suggestion
export default function ShippingMethods() {}
```
Another suggestions
Change `ShippingOptionsList` into `ShippingMethodOptions`
Change `ShipmentOptionsWithProducts` into `MultiAddreszShippingMethods`

Reply:
https://github.com/SalesforceCommerceCloud/pwa-kit/pull/3186